### PR TITLE
Update subtitles to use config.php dir locations

### PR DESCRIPTION
**Pull request by TTsWeb**
Updated subtitles.php and downloadsub.php to use config.php location for the dirs.
Should make it easier for people to locate the dirs instead of in separate files.
Not my work came from v5 I am just pushing it into v4 :)

### CODE REVIEW

File: subtitles.php [hunk 1]
```diff
@@ -120,7 +119,7 @@ function htmlsafechars($var)
                 if ($arr["fps"] != $fps) $updateset[] = "fps = " . sqlesc($fps);
                 if ($arr["cds"] != $cd) $updateset[] = "cds = " . sqlesc($cd);
                 if ($arr["comment"] != $comment) $updateset[] = "comment = " . sqlesc($comment);
-                if (count($updateset) > 0) sql_query("UPDATE subtitles SET " . join(",", $updateset) . " WHERE id ={$id} ") or sqlerr(__FILE__, __LINE__);
+                if (count($updateset) > 0) sql_query("UPDATE subtitles SET " . join(",", $updateset) . " WHERE id =".sqlesc($id)) or sqlerr(__FILE__, __LINE__);
```
Comment (ChubbyNinja) [inline subtitles.php]: No need to run `sqlesc` as you're forcing integer on line 107

File: subtitles.php [hunk 2]
```diff
@@ -105,10 +104,10 @@ function htmlsafechars($var)
             header("Refresh: 0; url=subtitles.php?mode=details&id=$id");
         } //end upload
         if ($action == "edit") {
-            $id = isset($_POST["id"]) ? 0 + $_POST["id"] : 0;
+            $id = isset($_POST["id"]) ? (int) $_POST["id"] : 0;
             if ($id == 0) stderr("Err", "Not a valid id");
             else {
-                $res = sql_query("SELECT * FROM subtitles WHERE id={$id} ") or sqlerr(__FILE__, __LINE__);
+                $res = sql_query("SELECT * FROM subtitles WHERE id=".sqlesc($id)) or sqlerr(__FILE__, __LINE__);
```
Comment (ChubbyNinja) [inline subtitles.php]: No need to run `sqlesc` as you're forcing integer on line 107

File: subtitles.php [hunk 3]
```diff
@@ -130,16 +129,16 @@ function htmlsafechars($var)
 } //end POST
 if ($mode == "upload" || $mode == "edit") {
     if ($mode == "edit") {
-        $id = isset($_GET["id"]) ? 0 + $_GET["id"] : 0;
+        $id = isset($_GET["id"]) ? (int) $_GET["id"] : 0;
         if ($id == 0) stderr("Err", "Not a valid id");
         else {
-            $res = sql_query("SELECT id, name, imdb, poster, fps, comment, cds, lang FROM subtitles WHERE id={$id} ") or sqlerr(__FILE__, __LINE__);
+            $res = sql_query("SELECT id, name, imdb, poster, fps, comment, cds, lang FROM subtitles WHERE id=".sqlesc($id)) or sqlerr(__FILE__, __LINE__);
```
Comment (ChubbyNinja) [inline subtitles.php]: No need to run `sqlesc` as you're forcing integer on line 132

**Comment by ChubbyNinja** [comment]
Multiple times you're forcing integer with `(int)$` but still using `sqlesc($id)`

`sqlesc` is not needed for integers, so leaving as `{$id}` is fine

**Comment by Bigjoos** [comment]
Yeah but forcing integer will not necessarily prevent an sql injection, I'm sure i was able to break out the statement even after forcing (int) on that scripts queries or maybe im just going senile in my old age lmao, point taken though dude.

I'm looking at redoing everything with prepared statements but time is not a luxury i have right now unfortunately.

**Comment by ChubbyNinja** [comment]
Haha I'm sure you're not.. but the first line in the sqlesc function checks
for integer and just `return (int)` anyway.. so it's just double (int)
which makes the second one redundant.

Is v5 prepared statements?

**Comment by Bigjoos** [comment]
Not yet dude, will be after i release V5, I'm short on the ground with coders capable of helping me in that, but I'm working on that very issue

**Pull request merged**